src/api/uploads/FolderUploadNode.js

```diff
@@ -140,6 +146,60 @@ class FolderUploadNode {
             folderAPI.create(this.parentFolderId, this.name, resolve, reject);
         });
     }
+
+    /**
+     * Build current folder from entry
+     *
+     * @private
+     * @returns {Promise<any>}
+     */
+    buildCurrentFolderFromEntry = (): Promise<any> => {
+        if (!this.entry) {
+            return Promise.resolve();
+        }
+
+        const me = this;
+
+        return new Promise(async (resolve) => {
```

Comment:
Can the nesting here be reduced? By moving stuff at the root level? Say for the read() function.